storage/reads/stream_reader.go

```diff
@@ -44,23 +45,28 @@ func (*emptyStatistics) Stats() cursors.CursorStats {
 	return cursors.CursorStats{}
 }
 
-// StorageReadClient adapts a Storage_ReadClient to implement cursors.Statistics
-// and read the statistics from the gRPC trailer.
+type streamClient interface {
+	StreamReader
+	grpc.ClientStream
+}
+
+// StorageReadClient adapts a grpc client to implement the cursors.Statistics
+// interface and read the statistics from the gRPC trailer.
 type StorageReadClient struct {
-	c       datatypes.Storage_ReadClient
+	client  streamClient
 	trailer metadata.MD
 }
 
 // NewStorageReadClient returns a new StorageReadClient which implements
 // StreamReader and reads the gRPC trailer to return CursorStats.
-func NewStorageReadClient(c datatypes.Storage_ReadClient) *StorageReadClient {
-	return &StorageReadClient{c: c}
+func NewStorageReadClient(client streamClient) *StorageReadClient {
```

Comment:
This is an exported API, so I would expect `streamClient` to be exported as `StreamClient` for correctness.